wagtail/snippets/bulk_actions/snippet_bulk_action.py

```diff
@@ -0,0 +1,27 @@
+from django.urls import reverse
+
+from wagtail.admin.views.bulk_action import BulkAction
+from wagtail.snippets.models import get_snippet_models
+
+
+def get_edit_url(app_label, model_name, snippet):
+    return reverse(f"wagtailsnippets_{app_label}_{model_name}:edit", args=[snippet.pk])
```

Comment:
Snippets can have custom primary keys, so we must quote them to escape characters like `/` in the URL:

```suggestion
    # from django.contrib.admin.utils import quote
    return reverse(f"wagtailsnippets_{app_label}_{model_name}:edit", args=[quote(snippet.pk)])
```